website/docs/docs/building-a-dbt-project/hooks-operations.md

```diff
@@ -27,14 +27,117 @@ Effective database administration sometimes requires additional SQL statements t
 
 dbt provides two different interfaces for you to version control and execute these statements as part of your dbt project — hooks and operations.
 
-### Hooks
+## About hooks
+
 Hooks are snippets of SQL that are executed at different times:
   * `pre-hook`: executed _before_ a model, seed or snapshot is built.
   * `post-hook`: executed _after_ a model, seed or snapshot is built.
   * `on-run-start`: executed at the _start_ of `dbt run`, `dbt seed` or `dbt snapshot`
   * `on-run-end`: executed at the _end_ of `dbt run`, `dbt seed` or `dbt snapshot`
 
-Hooks are defined in your `dbt_project.yml` file. Pre- and post-hooks can also be defined in a `config` block.
+Hooks can help you leverage database-specific actions to expand dbt.
+
+You can define hooks in your `dbt_project.yml` file. Pre- and post-hooks can also be defined in a `config` block.
+
+<VersionBlock firstVersion="1.2">
+
+In order to streamline hooks and automatically apply grants when your dbt model runs, we recommend using [`grants` resource-config](/reference/resource-configs/grants).  
+
+### Examples using hooks
+
+You can use hooks to trigger actions at certain times when running an operation or building a model, seed, or snapshot.
+
+For more information about when hooks can be triggered, see reference sections for [`on-run-start` and `on-run-end` hooks](on-run-start-on-run-end) and [`pre-hook`s and `post-hook`s](pre-hook-post-hook).
+
+You can use hooks to provide database-specific functionality not available out-of-the-box with dbt. 
+
+For example, you can use a `config` block to run an `ALTER TABLE` statement right after building an individual model using a `post-hook`:
+
+<WHCode>
```

Comment:
I left some specific comments correcting the syntax on each warehouse, but I also want to be clear that all the examples below are _simple grant statements_, and we should be encouraging anyone on v1.2+ to use the `grants` config instead.

We should explicitly discourage them from using `post-hook` for simple grants. Instead, they should use `post-hook` if and only if:
- they need to apply grants in a more complex way, which the v1.2 `grants` config does not (yet) support
- they need to do other custom post-processing that dbt does not (yet) support via out-of-the-box configuration — e.g. [`analyze table`](https://spark.apache.org/docs/latest/sql-ref-syntax-aux-analyze-table.html), [`alter table set property`](https://spark.apache.org/docs/latest/sql-ref-syntax-ddl-alter-table.html), [`alter table ... add row access policy`](https://docs.snowflake.com/en/sql-reference/sql/alter-table.html), etc

Reply:
Added here: https://github.com/dbt-labs/docs.getdbt.com/pull/1642/commits/44e2ef211cb53d609332aedd7eec5e7e2e4af5d5 but need to edit more

Reply:
It's still a bit weird to see the `post-hook` examples below all featuring simple `grant` statements, when we've explicitly said above that users should NOT use `post-hook` to achieve that functionality.

It would make more sense to me if the section below was `<VersionBlock lastVersion="1.1">`, and then for v1.2+, we just had a generic placeholder post-hook (not `WHCode`) like:
```sql
{{ config(
    post_hook=[
      "alter table {{ this }} ..."
    ]
) }}
```

Reply:
I think I got it right this time. Added here: https://github.com/dbt-labs/docs.getdbt.com/pull/1642/commits/668e2472672a32838b5544d29dfff2e28f70bf06